be/src/runtime/hdfs/hdfs_fs_cache.cpp

```diff
@@ -44,18 +71,28 @@ static Status create_hdfs_fs_handle(const std::string& namenode, HdfsFsHandle* h
 Status HdfsFsCache::get_connection(const std::string& namenode, HdfsFsHandle* handle, const FSOptions& options) {
     {
         std::lock_guard<std::mutex> l(_lock);
-        std::string id = namenode;
+        std::string cache_key = namenode;
         const THdfsProperties* properties = options.hdfs_properties();
         if (properties != nullptr && properties->__isset.hdfs_username) {
-            id += properties->hdfs_username;
+            cache_key += properties->hdfs_username;
         }
-        auto it = _cache.find(id);
+
+        // Insert azure cloud credential into cache key
+        const std::vector<TCloudProperty>* azure_cloud_properties = get_azure_cloud_properties(options);
```

Comment:
why only add azure? how about aws?

Reply:
AWS uses s3 sdk in BE, it will not go through here. `hdfs_fs_cache.cpp` is only used by protocol which using LibHDFS.